xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/DocumentTagsTest.java

```diff
@@ -143,4 +144,41 @@ void displayTagsWhenNoEditRightsAndTagPluginAvailableAndTags() throws Exception
             + "</span> "
             + "</div>"));
     }
+
+    @Test
+    void displayTagsWhenEditRightsAndTagPluginAvailableAndTags() throws Exception
+    {
+        // Give edit rights ($hasEdit = true)
+        when(this.oldcore.getMockContextualAuthorizationManager().hasAccess(Right.EDIT)).thenReturn(true);
+
+        // Add tags to the current document
+        XWikiDocument currentDocument = this.context.getDoc();
+        BaseObject baseObject = new BaseObject();
+        baseObject.setXClassReference(new DocumentReference("xwiki", "XWiki", "TagClass"));
+        baseObject.setStringListValue("tags", Arrays.asList("tag1", "tag2"));
+        currentDocument.addXObject(baseObject);
+        this.xwiki.saveDocument(currentDocument, this.context);
+
+        TemplateManager templateManager = this.oldcore.getMocker().getInstance(TemplateManager.class);
+        // Remove extra spaces to make it easy to assert the result below.
+        String result = templateManager.render("documentTags.vm").trim().replaceAll("\\s+", " ");
+
+        // Verify that the generated HTML matches the expectations:
+        // - The tag label is displayed
+        // - The tags after the tag label
+        // - The "+" link is displayed since the user has edit rights
+        assertThat(result, matchesPattern("\\Q<div class=\"doc-tags\" id=\"xdocTags\"> core.tags.list.label "
+        + "<span class=\"tag-wrapper\"> <span class=\"tag\">"
+        + "<a href=\"/xwiki/bin/view/Main/Tags?do=viewTag&amp;tag=tag1\">tag1</a></span> "
+        + "<span class=\"separator\">[</span>"
+        + "<a href=\"/xwiki/bin/view/space/page?xpage=documentTags&amp;xaction=delete&amp;tag=tag1&amp;form_token=&amp;xredirect=%2Fxwiki%2Fbin%2Fview%2Fspace%2Fpage%23xdocTags\" "
```

Comment:
I'm wondering if it really makes sense to assert the full URL or if this wouldn't be a good place to actually use the fact that you're in a regular expression and just, e.g., match `[^"]*` inside the URL (at least after some point, you could still match, e.g., the start of at least the first tag's URL).